main/webapp/modules/core/scripts/dialogs/clustering-dialog.js

```diff
@@ -256,23 +272,28 @@ ClusteringDialog.prototype._renderTable = function(clusters) {
                 editCheck.attr("checked", "true");
             }
 
-            var input = $('<input type="text" size="25" />')
+            $('<input type="text" size="25" />')
                 .attr("value", cluster.value)
                 .bind("keyup change input",function() {
                     cluster.value = this.value;
                 }).appendTo(tr.insertCell(4));
+
+            return choices.length;
         };
 
-        for (var i = 0; i < clusters.length; i++) {
-            renderCluster(clusters[i]);
+        var maxRenderRows = 10000;
```

Comment:
@wetneb If you want to experiment with the number of rows that your machine can handle, the limit is set here